conformance/tests/httproute-additional-redirect-scheme.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/roundtripper"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/pkg/features"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteAdditionalRedirectScheme)
+}
+
+var HTTPRouteAdditionalRedirectScheme = suite.ConformanceTest{
+	ShortName:   "HTTPRouteAdditionalRedirectScheme",
+	Description: "An HTTPRoute with a additional scheme redirect filter",
+	Manifests:   []string{"tests/httproute-additional-redirect-scheme.yaml"},
+	Features: []features.FeatureName{
+		features.SupportGateway,
+		features.SupportHTTPRoute,
+		features.SupportHTTPRouteSchemeRedirect,
+		features.SupportHTTPRouteAdditionalRedirectStatusCodes,
```

Comment:
are we anticipating any more codes in the future? haha. 
should we specifically mention 307 and 308 in the feature name or no need?

Reply:
> I'm not opposed to this, although it feels like it might be excessively verbose for what I expect would just be adding a few additional allowed values to implementations and not really require any substantial additional logic (or discrete changes for each different code).

Agreed. But in the odd case that we get to a point in time where we want to make core only a subset of these, having different features makes our life easier. I agree this is more verbose, but I don't see much harm in it, honestly.